core/src/main/java/tc/oc/pgm/kits/KitParser.java

```diff
@@ -398,6 +405,47 @@ public ItemStack parseHead(Element el) throws InvalidXMLException {
     return itemStack;
   }
 
+  public ItemStack parseFirework(Element el) throws InvalidXMLException {
+    ItemStack itemStack = parseItem(el, Material.FIREWORK);
+    FireworkMeta meta = (FireworkMeta) itemStack.getItemMeta();
+    int power = XMLUtils.parseNumber(Node.fromAttr(el, "power"), Integer.class, false, 1);
+    meta.setPower(power);
+
+    for (Element explosionEl : el.getChildren("explosion")) {
+      Type type =
+          XMLUtils.parseEnum(Node.fromAttr(explosionEl, "type"), Type.class, null, Type.BURST);
+      boolean flicker = XMLUtils.parseBoolean(Node.fromAttr(explosionEl, "flicker"), false);
+      boolean trail = XMLUtils.parseBoolean(Node.fromAttr(explosionEl, "trail"), false);
+
+      List<Color> primary = parseColors(Node.fromChildren(explosionEl, "color"));
+      List<Color> fade = parseColors(Node.fromChildren(explosionEl, "fade"));
+
+      if (primary.isEmpty()) {
+        throw new InvalidXMLException("At least one <color> must be defined", explosionEl);
+      }
+
+      meta.addEffect(
+          FireworkEffect.builder()
+              .with(type)
+              .withColor(primary)
+              .withFade(fade)
+              .flicker(flicker)
+              .trail(trail)
+              .build());
+    }
+
+    itemStack.setItemMeta(meta);
+    return itemStack;
+  }
+
+  private List<Color> parseColors(List<Node> nodes) throws InvalidXMLException {
+    List<Color> colors = Lists.newArrayList();
```

Comment:
```suggestion
    List<Color> colors = new ArrayList(nodes.size());
```
pre-allocating the size makes alot of sense in this scenario